### PR TITLE
Create an API to get upcoming events

### LINKED ISSUE #12: Create an API to get upcoming events
**Is your feature request related to a problem? Please describe.**
Upcoming onelive events should be displayed on the sef-site. For that, an API to get upcoming events from Google Calendar API is needed.

**Describe the solution you'd like**
Add an endpoint to get google calendar event list

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Piumal1999**
## Purpose

The purpose of this PR is to fix #12 

## Goals

Create an API to get upcoming events from Google Calendar

## Approach

Create an API endpoint to get upcoming events using the Google Calendar API.

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Related PRs
 
N/A

## Test environment
 
Ubuntu 19.04
openjdk version 11.0.5

## Learning

https://developers.google.com/calendar/v3/reference/events/list?apix_params=%7B%22calendarId%22%3A%22ijucjv32furo2ue6bfqg16lh60%40group.calendar.google.com%22%7D

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,9 +8,15 @@
     ```
 - Set Youtube API key and Youtube playlist ID as environment variables
     ```
-    export YOUTUBE_API_KEY==your_youtube_api_key
+    export YOUTUBE_API_KEY=your_youtube_api_key
```
Comment (jayasanka-sack) [inline README.md]: The youtube API key and Google API key is the same. 

So remove this environment variable and keep only the `GOOGLE_API_KEY` 
Change @anjula-sack 's implementation according to this.

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jayasanka-sack) [reply README.md]: Also please mention that the calendar has to be public

Comment (Piumal1999) [reply README.md]: Resolved

File: README.md [hunk 2]
```diff
@@ -6,11 +6,18 @@
...
-- Set Youtube API key and Youtube playlist ID as environment variables
+- Set Google API key and Youtube playlist ID as environment variables
     ```
-    export YOUTUBE_API_KEY==your_youtube_api_key
+    export GOOGLE_API_KEY=your_google_api_key
     export YOUTUBE_PLAYLIST_ID=your_playlist_id
     ```
+- Set Google API key and Google Calendar ID as environment variables to use the Calendar API
+    ```
+    export GOOGLE_API_KEY=your_google_api_key
```
Comment (jayasanka-sack) [inline README.md]: This has been mentioned twice.

File: src/main/java/org/sefgobal/dataholder/api/CalendarAPI.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import java.nio.charset.StandardCharsets;
+
+
+@RestController
+public class CalendarAPI {
+
+    @GetMapping("/calendar/onelive")
+    public void getUpcomingOnelives(HttpServletRequest request, HttpServletResponse response) {
+        String googleApiKey = System.getenv("GOOGLE_API_KEY");
+        String calendarId = System.getenv("CALENDAR_ID");
```
Comment (jayasanka-sack) [inline src/main/java/org/sefgobal/dataholder/api/CalendarAPI.java]: Let's rename this as `ONELIVE_CALENDAR_ID`

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (jayasanka-sack) [reply README.md]: Add a link to a documentation/tutorial on how to create an API key.

Mention that the user has to enable the relevant APIs for the project in google console. ( Youtube, Calendar )

**Review by anjula-sack — DISMISSED** [review]
LGTM

File: src/main/java/org/sefgobal/dataholder/api/CalendarAPI.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+
+@RestController
+public class CalendarAPI {
+
+    @GetMapping("/calendar/onelive")
+    public void getUpcomingOnelives(HttpServletRequest request, HttpServletResponse response) {
+        String googleApiKey = System.getenv("GOOGLE_API_KEY");
+        String oneliveCalendarId = System.getenv("ONELIVE_CALENDAR_ID");
+        String url = "https://www.googleapis.com/calendar/v3/calendars/"+oneliveCalendarId+"/events?key="+googleApiKey;
```
Comment (jayasanka-sack) [inline src/main/java/org/sefgobal/dataholder/api/CalendarAPI.java]: One last thing. 
Format this file

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]

**Review by jayasanka-sack — APPROVED** [review]
Perfect!

**Pull request merged**